File: app/src/main/java/com/vibhorsrv/cameraids/util/CameraUtil.java
package com.vibhorsrv.cameraids.util;

import android.util.Size;
import android.util.SizeF;

public class CameraUtil {
    public static float calculatePixelSize(int pixelArrayWidth, float sensorWidth) {
        return (sensorWidth / ((float) pixelArrayWidth)) * 1000.0f;
    }

    public static Double calculateAngleOfView(float focalLength, SizeF sensorSize, Size pixelArraySize) {
        float pixelSize = CameraUtil.calculatePixelSize(pixelArraySize.getWidth(), sensorSize.getWidth());
        return Math.toDegrees(Math.atan(Math.sqrt(Math.pow(sensorSize.getWidth() * pixelSize, 2.0d)
                + Math.pow(sensorSize.getHeight() * pixelSize, 2.0d)) / ((double) (2.0f * focalLength))) * 2.0d);
    }

    public static float calculate35mmeqv(float focalLength, SizeF sensorSize) {
        return (36.0f / sensorSize.getWidth()) * focalLength;
    }

}

File: app/src/main/java/com/vibhorsrv/cameraids/util/CameraUtil.kt
package com.vibhorsrv.cameraids.util

import android.util.Size
import android.util.SizeF
import kotlin.math.atan
import kotlin.math.pow
import kotlin.math.sqrt

object CameraUtil {
    fun calculatePixelSize(pixelArrayWidth: Int, sensorWidth: Float): Float {
        return sensorWidth / pixelArrayWidth.toFloat() * 1000.0f
    }

    fun calculateAngleOfView(
        focalLength: Float, sensorSize: SizeF, pixelArraySize: Size
    ): Double {
        val pixelSize = calculatePixelSize(pixelArraySize.width, sensorSize.width)
        return Math.toDegrees(atan(sqrt((sensorSize.width * pixelSize).toDouble().pow(2.0)
                + (sensorSize.height * pixelSize).toDouble().pow(2.0)
        ) / (2.0f * focalLength).toDouble()) * 2.0)
    }

    fun calculate35mmeqv(focalLength: Float, sensorSize: SizeF): Float {
        return 36.0f / sensorSize.width * focalLength
    }
}
